frontends/concrete-python/concrete/fhe/compilation/server.py

```diff
@@ -416,10 +440,41 @@ def run(
                 raise ValueError(message)
 
             if not isinstance(arg, Value):
-                message = f"Expected argument {i} to be an fhe.Value but it's {type(arg).__name__}"
-                raise ValueError(message)
+                if i not in self._clear_input_indices[function_name]:
+                    message = (
+                        f"Expected argument {i} to be an fhe.Value "
+                        f"but it's {friendly_type_format(type(arg))}"
+                    )
+                    raise ValueError(message)
+
+                exporter = SimulatedValueExporter.new(
+                    self.client_specs.client_parameters,
+                    function_name,
+                )
 
-            buffers.append(arg.inner)
+                if isinstance(arg, (int, np.integer)):
+                    shape = ()
+                else:
+                    arg = np.array(arg)
+                    shape = arg.shape
+
+                expected_shape = self._clear_input_shapes[function_name][i]
+                if shape != expected_shape:
```

Comment:
Is that not already checked by the exporter?

Reply:
It was but I didn't like the error message. Anyway I used the builtin check to not repeat ourselves.